framework/include/base/DependencyResolverInterface.h

```diff
@@ -72,6 +90,9 @@ DependencyResolverInterface::sort(typename std::vector<T> & vector)
 
   for (typename std::vector<T>::iterator iter = start; iter != end ; ++iter)
   {
+    if ((*iter)->allowCyclicDepenedency())
+      continue;
```

Comment:
The first three bullets actually makes this interface more consistent with other interfaces, most have parameters.

Bypassing the dependency check was the simpliest way to get the `DiscreteMaterials` to work as intended without doing one of the following:
    - creating a separate storage structure for `DiscreteMaterial` objects, but doing this will require creating separate `MaterialData` objects so that all the material property checks can be handled properly. 
   - another option is to create separate storage contains, but re-join them for the depedency checks, but this involves alot of `if (discrete)` stuff all over the place. I started down this road initially and it just seemed messy. At this point the `DiscreteMaterial` should just be its own system, but that leads back to the first issue of MaterialData.